File: Java/oop/OOP/src/belajar/java/data/Product.java
package belajar.java.data;

import java.util.Objects;

public class Product {
    private String name;
    private int price;

    public Product(String name, int price) {
        this.name = name;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public int getPrice() {
        return price;
    }

    public String toString() {
        return "Product name:" + name + ", price:" + price;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;

        Product product = (Product) o;
        return price == product.price && Objects.equals(name, product.name);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(name);
        result = 31 * result + price;
        return result;
    }

//    public boolean equals(Object o) {
//        if (o == this)
//            return true;
//        if (!(o instanceof Product))
//            return false;
//
//        Product product = (Product) o;
//
//        if (this.price != product.price){
//            return false;
//        }
//
//        if (this.name != null){
//            return this.name.equals(product.name);
//        } else {
//            return product.name == null;
//        }
//    }

}

File: programming-language/PHP/programerzamannow/pembelajaran/oop/OOP/data/Product.php
<?php

class Product
{
    private string $name;
    private int $price;
    protected ?string $description;

    public function __construct(string $name, int $price, string $description = "No Description")
    {
        $this->name = $name;
        $this->price = $price;
        $this->description = $description;
    }

    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getPrice(): int
    {
        return $this->price;
    }
}

class ProductDescription extends Product
{
    public function Description(): void
    {
        echo "Name: " . $this->getName() . PHP_EOL;
        echo "Price: " . $this->getPrice() . PHP_EOL;
        echo "Description: " . $this->description . PHP_EOL;
    }
}